script/test-robots

```diff
@@ -0,0 +1,141 @@
+import re
+import sys
+import time
+import shutil
+import argparse
+import tempfile
+import subprocess
+from typing import List, Optional
+from pathlib import Path
+from zipfile import ZipFile
+
+REPO_ROOT = Path(__file__).resolve().parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+from controller_utils import NUM_ZONES  # isort:skip
+
+EXCESS_SUBPROCESS_TIMEOUT = 10
+
+
+def run_robot_match(
+    comp_match_script: Path,
+    archives_dir: Path,
+    tla: str,
+    failed_tlas: Optional[List[str]],
+    subprocess_timeout: Optional[int],
+) -> int:
+    try:
+        print(f"Running match for {tla}")  # noqa:T001
+
+        match_args = [
+            sys.executable,
+            str(comp_match_script),
+            str(archives_dir),
+            '0',  # no logs will be retained so a fixed match number is used
+        ]
+
+        match_args.append(tla)
+
+        match_args.extend(['-'] * (NUM_ZONES - 1))  # all other zones are empty
+
+        match_args.extend([
+            '--duration', '1',
+            '--no-record',
+        ])
+
+        start_time = time.time()
+        subprocess.run(
+            match_args,
+            check=True,
+            stdout=subprocess.PIPE,  # we only want to print output from failing robots
+            stderr=subprocess.STDOUT,
+            timeout=subprocess_timeout,
+            text=True,
+        )
+
+        print(f"Completed match for {tla}")  # noqa:T001
+
+    except subprocess.CalledProcessError as match_errors:
+        print(match_errors.output)  # noqa: T001
+        print(f"Failed on team {tla}")  # noqa:T001
+        if failed_tlas is None:
+            raise match_errors
+        else:
+            failed_tlas.append(tla)
+    except subprocess.TimeoutExpired as match_timeout:
+        print(match_timeout.output.decode())  # noqa: T001
+        print(f"Timeout on team {tla}")  # noqa:T001
+        if failed_tlas is not None:
+            failed_tlas.append(tla)
+    return round(time.time() - start_time)
+
+
+def is_team_archive(archive: str) -> bool:
+    # team archives are three uppercase letters and an optional number
+    return bool(re.search(r'^[A-Z]{3}[0-9]?$', archive))
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'archives_dir',
+        help=(
+            "The directory, or archive, containing the teams' robot code, "
+            "as Zip archives named for the teams' TLAs."
+        ),
+        type=Path,
+    )
+    return parser.parse_args()
+
+
+def main(args: argparse.Namespace) -> None:
+    failed_tlas: List[str] = []
+
+    run_comp_match = Path(__file__).parent / 'run-comp-match'
+
+    with tempfile.TemporaryDirectory() as tempdir_name:
+
+        tempdir = Path(tempdir_name)
+        # Make a temporary copy of the archives to receive the logs
+        if args.archives_dir.is_dir():
+            archives = shutil.copytree(args.archives_dir.resolve(), tempdir / 'archives')
```

Comment:
Is `.resolve()` here useful?

Reply:
Not really.

Reply:
Removed in 86bea565f705f65a3f0e1cfe2155837501081991